### PR TITLE
document that the custom event category and action properties are optional

### PR DESCRIPTION
**Pull request by JakeChampion**
fixes https://github.com/Financial-Times/o-tracking/issues/72

### CODE REVIEW

File: src/javascript/events/custom.js [hunk 1]
```diff
@@ -30,12 +18,14 @@ const defaultEventConfig = function () {
...
- *   [component_id] - Optional. The ID for the component instance.
+ * @param {Object} trackingEvent - The event, which could the following properties in its 'detail' key:
+ * @param {Object} [trackingEvent.detail] - Custom properties to add to the event.
+ * @param {string} [trackingEvent.detail.category] - Category for this event e.g. page
+ * @param {string} [trackingEvent.detail.action] - Action for this event e.g. view
+ * @param {string} [trackingEvent.detail.component_id] - The ID for the component instance.
+ * @param {Object} [trackingEvent.detail.context] - Extra context to add to the event
  *
- * @param {Function} callback - Optional, Callback function. Called when request completed.
+ * @param {Function=} callback - Optional, Callback function. Called when request completed.
```
Comment (JakeChampion) [inline src/javascript/events/custom.js:28]:
```suggestion
 * @param {Function} [callback] - Optional, Callback function. Called when request completed.
```

File: README.md [hunk 2]
```diff
@@ -227,8 +227,8 @@ oTracking.view.init(opts);
 To track a custom event call the `oTracking.event` method:
```
Comment (notlee) [inline README.md:227]:
```suggestion
To track a custom event call the `oTracking.event` method. For example to track that a video is played:
```

**Review by notlee — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -227,8 +227,8 @@ oTracking.view.init(opts);
...
 ```js
 const eventConfig = {
-    "category": "video",
-    "action": "play",
+    "category": "video", // optional
```
[the remark this answers is not in the document]
Comment (JakeChampion) [reply README.md:230]: Nope :D

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 4556d506ea73fe2b4a00265b6b651aa6f5a2e1d4
js: 5.4kb increase (1.64kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to a15dc0b03ede5220576df32ebb9e83b39a649862
js: 5.4kb increase (1.64kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to fc72f36e62fae5757e650736a20590cbc31d817b
js: 5.4kb increase (1.64kb/gzip)

**Comment by chrisbrownuk** [comment]
To date category and action have always been treated as compulsory.  At least one value should exist, at least, otherwise no one will know what the event is!  

@jaya-ananthram-ft, @stefkaargirova We need to check on potential downstream consequences of either category or action being omitted.

**Comment by JakeChampion** [comment]
They have always been optional in o-tracking users because o-tracking sets default values for both

**Pull request merged**